apps/main/src/lend/entities/chain/chain-hooks.ts

```diff
@@ -1,20 +1,21 @@
 import { useMemo } from 'react'
 import networks from '@/lend/networks'
-import { ChainId } from '@/lend/types/lend.types'
+import { type Api, ChainId } from '@/lend/types/lend.types'
 import { OneWayMarketTemplate } from '@curvefi/lending-api/lib/markets'
+import { useConnection } from '@ui-kit/features/connect-wallet'
+import { isHydrated } from '@ui-kit/features/connect-wallet/lib/ConnectionContext'
 import { ChainParams } from '@ui-kit/lib/model/query'
-import { useApiStore } from '@ui-kit/shared/useApiStore'
 import { useOneWayMarketNames } from './chain-query'
 
 export const useOneWayMarketMapping = (params: ChainParams<ChainId>) => {
   const { chainId } = params
   const { data: marketNames, ...rest } = useOneWayMarketNames(params)
-  const api = useApiStore((state) => state.lending)
-  const isLoadingApi = useApiStore((state) => state.isLoadingLending)
+  const { lib: api, connectState } = useConnection<Api>()
   const apiChainId = api?.chainId
   const data: Record<string, OneWayMarketTemplate> | undefined = useMemo(
     () =>
-      marketNames && api && chainId == apiChainId && !isLoadingApi
+      // note: only during hydration `api` internally retrieves all the markets, and we can call `getOneWayMarket`
+      isHydrated(connectState) && marketNames && api && chainId == apiChainId
```

Comment:
Won't this temporarily break old data while re-hydrating because of the 5m periodic refresh timer? So one moment you have data, but while it's refreshing you temporarily have no data. I don't think this is a blocker though, I suspect you'll just see a loader spinner somewhere for a brief moment.

Reply:
There is a perfetch during hydration. Pretty sketchy tho 😢